src/lib/challenge.ts

```diff
@@ -48,3 +52,26 @@ export async function generate (size: number = 32): Promise<string> {
     throw error
   }
 }
+
+/**
+ * Helper function to validate signatures using public key
+ * @param challenge UTF-8 challenge string
+ * @param signature Base64 sign string
+ * @param publicKey Base64 RSA/EC:secp256k1 key string or KeyObject for verification
+ */
+export function verifySign (
+  challenge: string,
+  signature: string,
+  publicKey: string | crypto.KeyObject): boolean {
+  try {
+    // Hash using SHA256
+    const verifier: crypto.Verify = crypto.createVerify('SHA256')
+
+    verifier.update(challenge)
+
+    return verifier.verify(publicKey, signature, 'base64')
+  } catch (error) {
+    Logger.push(error).error('Unable to verify signature')
```

Comment:
where this line of code is tested?

Reply:
Could you suggest something here @eoln ? I was thinking of adding a test for this but I didn't know how to get the function to throw an error here.

I just followed the pattern in the existing function from @akapur99 . There is no error test case for that.

Reply:
bad @akapur99 :-P 
@ramanmangla you should use jest.mock on `crypto` module with mocked function `createVerify` which throws an error so your catch is activated